influxql/iterator.go

```diff
@@ -539,3 +585,102 @@ type nilFloatIterator struct{}
 
 func (*nilFloatIterator) Close() error      { return nil }
 func (*nilFloatIterator) Next() *FloatPoint { return nil }
+
+// integerReduceSliceFloatIterator executes a reducer on all points in a window and buffers the result.
+// This iterator receives an integer iterator but produces a float iterator.
+type integerReduceSliceFloatIterator struct {
```

Comment:
Instead of doing this, an idea that I had for implicit float conversions was to have a special cast iterator. The cast iterator would just read integer points from an input iterator and act as a float iterator by outputting float points. Doing this, it extends to binary expressions (which will also need this same functionality) and to aggregate functions.

Thoughts?